crates/proto/src/rr/domain/label.rs

```diff
@@ -63,9 +63,14 @@ impl Label {
         match idna::Config::default()
             .use_std3_ascii_rules(true)
             .transitional_processing(true)
-            .verify_dns_length(true)
+            // checked manually after it's okay domain is already processed.
+            // and Errors is... opaque.
+            .verify_dns_length(false)
             .to_ascii(s)
         {
+            Ok(puny) if puny.len() > 63 => {
+                return Err(ProtoErrorKind::LabelBytesTooLong(puny.len()).into())
+            }
```

Comment:
Made this to have a better error than just the opaque tipe of `idna` crate they do something like this:

SRC:
https://github.com/servo/rust-url/blob/master/idna/src/uts46.rs#L490-L493
```
            if domain.len() > 253 || domain.split('.').any(|label| label.len() > 63) {
                errors.too_long_for_dns = true;
            }
```